src/Lean/Elab/Notation.lean

```diff
@@ -53,16 +53,17 @@ def mkSimpleDelab (attrKind : Syntax) (vars : Array Syntax) (pat qrhs : Syntax)
     guard <| args.all (Syntax.isIdent ∘ getAntiquotTerm)
     guard <| args.allDiff
     -- replace head constant with (unused) antiquotation so we're not dependent on the exact pretty printing of the head
+    -- The reference is attached to the syntactic representation of the called function itself, not the entire function application
     `(@[$attrKind:attrKind appUnexpander $(mkIdent c):ident]
       aux_def unexpand $(mkIdent c) : Lean.PrettyPrinter.Unexpander := fun
-       | `($$(_):ident $args*) => `($pat)
-       | _                     => throw ())
+       | `($$fun_ref:ident $args*) => withRef fun_ref `($pat)
```

Comment:
That doesn't match our naming convention, let's go with just `f`? Same below. And mind the indentation.

Reply:
By the way, the *why* I was missing yesterday is that the quotation uses the reference as the source location of any newly introduced identifiers, which in this case is the function head.